Add `ignorePreludeOfAtRules: []` to `length-zero-no-unit` (#8472)
stylelint/stylelint: lib/rules/length-zero-no-unit/README.md

```css
a { top: --bar(0px); }
```

> ### `ignoreAtRulePreludes: ["/regex/", /regex/, "string"]`
> 
> Ignore units for zero lengths within the specified at-rule preludes.

Reviewer: ```suggestion
### `ignorePreludeOfAtRules: ["/regex/", /regex/, "string"]`

Ignore units for zero lengths within the preludes of the specified at-rules.
```

This is the first time we're adding an option that targets a _specific thing_ of the [_user-defined thing_](https://stylelint.io/developer-guide/rules#user-defined-ignore). As we typically end with the user-defined thing, e.g. `ignoreAtRules`, we can use a `<thing>Of` naming convention.

(Sorry for not considering in the issue and causing you to change it in the PR.)

Given:

```json

> ["function", "/^mix/"]

Reviewer: ```suggestion
["--foo", "/^--bar/"]
```

For "only use standard CSS syntax in examples and options" [convention](https://stylelint.io/developer-guide/rules#write-the-readme).

```

The following patterns are _not_ considered problems:

<!-- prettier-ignore -->
```css

> @mixin foo($x: 0px) { width: $x; }

Reviewer: ```suggestion
@--foo 0px {}
```

```

<!-- prettier-ignore -->
```css
@--bar-baz 0px;

> ```

Reviewer 2: [suggestion] Would you be willing to add a more practical example? E.g.

```json
"ignorePreludeOfAtRules": ["media"]
```

```css
@media (height > 0px) {}
```

Reviewer: Let's replace `--foo` with `media` and keep the `^--bar` custom at-rule example to show both use cases.

> @function baz($x: 0px) { @return $x; }

Reviewer: ```suggestion
@--bar-baz 0px;
```
